.gitignore

```diff
@@ -14,3 +14,6 @@ build/
 
 # NPM package generated files:
 dist/contracts.json
+
+# Do not track generated wrapper, it should be part of build script
+interacts/*
```

Comment:
I can see that you have added `postinstall` script.
Its mentioned in the typechain documentation not to commit the interacts. What will be the disadvantages if the interacts are committed?

I am thinking in general, If we publish an NPM package, will the interacts be available? Or the project that consumes the package needs to handle the interact generation separately?

Reply:
The interacts would still be part of the published package in my opinion, just not part of the repository. That way, it is guaranteed that the interacts are always up to date with the latest code changes on the solidity layer.

Reply:
Yes, Interacts will be available in the published package, it's just they are not tracked by git rather generated each time. This will make sure that everyone has the latest interacts. 

At the time of publishing, the latest interacts will be generated(as a build step)and published.

Reply:
```Yes, Interacts will be available in the published package, it's just they are not tracked by git rather generated each time.```

Wouldn't it be confusing on github if there are modules/libraries/helpers written on top of generated interacts 💭 ? Written modules will be tracked in git but the dynamic interacts will not be.
 
e.g. https://github.com/OpenST/openst.js/blob/develop/lib/helper/User.js
User.js is written on top of multiple contract interacts. So we will commit User.js in github calling generated interacts methods which will not be tracked in github.
One option is we describe this clearly in readme but I just feel it will be confusing for open source community looking at the code. 

What do you think 🤔 ?

Reply:
In the contract repository, there shouldn't be any complex class which uses multiple interacts. They should go into Interaction layer like mosaic.js, openst.js where these thin interact will always be available.